docs/source/asr/configs.rst

```diff
@@ -114,6 +114,41 @@ You can use any combination of Cutout, frequency/time SpecAugment, or none of th
 
 See the `Audio Augmentors <./api.html#Audio Augmentors>`__ API section for more details.
 
+Tokenizer Configurations
+------------------------
+
+Some models utilize sub-word encoding via an external tokenizer instead of explicitly defining their vocabulary.
+
+For such models, a ``tokenizer`` section is added  to the model config. ASR Models currently support two types of
+custom tokenizers.
+
+In order to build custom tokenizers, please refer to the ``ASR_with_Subword_Tokenization`` notebook available in the
+ASR tutorials directory.
+
+The following example sets up a ``SentencePiece Tokenizer`` at a path specified by the user:
+
+.. code-block:: yaml
+
+  model:
+    ...
+    tokenizer:
+      dir: "<path to the directory that contains the custom tokenizer files>"
+      type: "bpe"  # can be "bpe" for `sentencepiece` tokenizers or "wpe" for wordpiece tokenizers.
+
+For models which utilize sub-word tokenization, we share the decoder module (``ConvASRDecoder``) with character tokenization models. All parameters are shared, but for models which utilize sub-word encoding, there are minor differences when setting up the config. For such models, the tokenizer is utilized to fill in the missing information when the model is constructed automatically.
+
+For example, a decoder config corresponding to a sub-word tokenization model would look like this:
+
+.. code-block:: yaml
+
+  model:
+    ...
+    decoder:
+      _target_: nemo.collections.asr.modules.ConvASRDecoder
+      feat_in: *enc_final
+      num_classes: -1  # filled with vocabulary size from tokenizer at runtime
+      vocabulary: []  # filled with vocabulary from tokenizer at runtime
+
```

Comment:
How about adding some hints on how and when they need to fill out num_classes or vocabulary, or what enc_final is referring to here?

Reply:
A short sentence what ConvASRDecoder is exactly?